autoload/emmet.vim

```diff
@@ -376,10 +376,21 @@ endfunction
 
 function! emmet#getFileType(...) abort
   let flg = get(a:000, 0, 0)
-  let type = ''
-
+  
   if has_key(s:emmet_settings, &filetype)
-    let type = &filetype
+    return &filetype
+  endif 
+
+  let pos = emmet#util#getcurpos()
+  let type = synIDattr(synID(pos[1], pos[2], 1), 'name')
+  if type =~? '^css\w'
+    let type = 'css'
+  elseif type =~? '^html\w'
+    let type = 'html'
+  elseif type =~? '^js\w' || '^javascript'
```

Comment:
@mattn using case-insensitive matching now that works for both "HTML" and "html", "CSS" and "css", etc... : - ]